### PR TITLE
docs: material ui theme guide

### PR DESCRIPTION
**Pull request by ozkalai**
In this PR, We have been creating a Material UI Theme Guide in the document with the following subtitles.

- [x] Material UI Theme
- [x] ThemeProvider
- [x] Passing the Theme to ThemeProvider
- [x] Overriding Variables
- [x] Create Custom Theme
- [x] Dark Mode
- [x] Notistack Snackbar with Theme

### CODE REVIEW

File: documentation/docs/guides-and-concepts/theme/material-ui.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+} from "@pankod/refine-mui";
+import dataProvider from "@pankod/refine-simple-rest";
+import routerProvider from "@pankod/refine-react-router-v6";
+
+import { PostsList, PostCreate, PostEdit } from "pages/posts";
+
+const App: React.FC = () => {
+    return (
+        // highlight-next-line
+        <ThemeProvider theme={theme}>
```
Comment (omeraplak) [inline documentation/docs/guides-and-concepts/theme/material-ui.md:37]:
```suggestion
        <ThemeProvider theme={theme}>
```
Where does `theme` import from? Shouldn't we also add `fontSmoothing`?

File: documentation/docs/guides-and-concepts/theme/material-ui.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+        <div class="control red"></div>
+        <div class="control orange"></div>
+        <div class="control green"></div>
+    </div>
+    <img src={overridedTheme} alt="Theme variable override" />
+</div>
+
+<br />
+
+You can also override [`typography`](https://mui.com/material-ui/customization/typography/) variable like this:
```
Comment (omeraplak) [inline documentation/docs/guides-and-concepts/theme/material-ui.md:218]:
```suggestion
You can also override [`typography`](https://mui.com/material-ui/customization/typography/) variable like this:
```
You can also change the Default Font Family. Refine uses the font [`Montserrat`](https://fonts.google.com/specimen/Montserrat) by default and you can change it like this:

File: documentation/docs/guides-and-concepts/theme/material-ui.md [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+                {children}
+            </ThemeProvider>
+        </ColorModeContext.Provider>
+    );
+};
+```
+
+</p>
+    </details>
+
```
Comment (biskuvit) [inline documentation/docs/guides-and-concepts/theme/material-ui.md:631]:
```suggestion
```

File: documentation/docs/guides-and-concepts/theme/material-ui.md [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+</p>
+    </details>
+    <br/>
+    <details>
+      <summary>
+       ColorModeContext
+      </summary>
+           <p>
+
+```tsx title="src/contexts/index.tsx"
```
Comment (omeraplak) [inline documentation/docs/guides-and-concepts/theme/material-ui.md:577]: Wouldn't it be better if we put the Context code block first?

**Review by omeraplak — APPROVED** [review]

**Review by biskuvit — APPROVED** [review]

File: documentation/docs/guides-and-concepts/theme/material-ui.md [hunk 5]
```diff
@@ -0,0 +1,723 @@
...
+        // highlight-next-line
+        <ThemeProvider theme={YOUR_THEME_OBJECT}>
+            <CssBaseline />
+            <GlobalStyles styles={{ html: { WebkitFontSmoothing: "auto" } }} />
+            <Refine
+                routerProvider={routerProvider}
+                dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
+                ReadyPage={ReadyPage}
+                Layout={Layout}
+                LoginPage={LoginPage}
```
Comment (biskuvit) [inline documentation/docs/guides-and-concepts/theme/material-ui.md:49]: We should remove the login page.

**Pull request merged**